tests/annotator/annotator.cpp

```diff
@@ -0,0 +1,2131 @@
+                             std::make_pair(libcellml::CellMLElement::UNITS, std::any(model->units("units3"))),
+                             std::make_pair(libcellml::CellMLElement::VARIABLE, std::any(c4v1)),
+                             std::make_pair(libcellml::CellMLElement::VARIABLE, std::any(c2v2)),
+                             std::make_pair(libcellml::CellMLElement::MAP_VARIABLES, std::any(libcellml::VariablePair(c2v2, c4v2))),
+                             std::make_pair(libcellml::CellMLElement::COMPONENT, std::any(model->component("component2")->component("component3"))),
+                             std::make_pair(libcellml::CellMLElement::ENCAPSULATION, std::any(model)),
+                         }},
+        {"duplicateId4", {
+                             std::make_pair(libcellml::CellMLElement::UNIT, std::any(libcellml::UnitItem(std::make_pair(model->units("units2"), 0)))),
+                             std::make_pair(libcellml::CellMLElement::COMPONENT, std::any(model->component("component1"))),
+                             std::make_pair(libcellml::CellMLElement::COMPONENT, std::any(model->component("component4"))),
+                             std::make_pair(libcellml::CellMLElement::MAP_VARIABLES, std::any(libcellml::VariablePair(c2v1, c3v1))),
+                             std::make_pair(libcellml::CellMLElement::VARIABLE, std::any(c2v1)),
+                             std::make_pair(libcellml::CellMLElement::MAP_VARIABLES, std::any(libcellml::VariablePair(c2v2, c4v2))),
+                             std::make_pair(libcellml::CellMLElement::RESET_VALUE, std::any(model->component("component2")->reset(0))),
+                         }},
+
+    };
+
+    libcellml::VariablePair pair;
+    libcellml::VariablePair reversePair;
+    libcellml::VariablePair testPair;
+
+    auto annotator = libcellml::Annotator::create();
+    annotator->setModel(model);
+
+    EXPECT_EQ(ids, annotator->duplicateIds());
+
+    for (const auto &id : ids) {
+        auto num = annotator->duplicateCount(id);
+        for (size_t index = 0; index < num; ++index) {
+            libcellml::AnyItem item = annotator->item(id, index);
+
+            EXPECT_EQ(expectedItems[id][index].first, item.first);
+
+            switch (item.first) {
+            case libcellml::CellMLElement::COMPONENT:
+            case libcellml::CellMLElement::COMPONENT_REF:
+                EXPECT_EQ(std::any_cast<libcellml::ComponentPtr>(expectedItems[id][index].second), std::any_cast<libcellml::ComponentPtr>(item.second));
+                break;
+            case libcellml::CellMLElement::CONNECTION:
+            case libcellml::CellMLElement::MAP_VARIABLES:
+                pair = std::any_cast<libcellml::VariablePair>(expectedItems[id][index].second);
+                testPair = std::any_cast<libcellml::VariablePair>(item.second);
+                // TODO not sure how best to test these.
```

Comment:
remove?

Reply:
Yes. Done it as part of my reviewing of this PR.